autosarfactory/autosar_ui.py

```diff
@@ -167,11 +217,28 @@ def __on_search_entry_click(self, event):
             self.__search_field.insert(0, '') #Insert blank for user input
             self.__search_field.config(foreground='black')
         elif search_string != '':
-            search_nodes = []
-            for node in self.__asr_explorer_id_to_node_dict.values():
-                if node.name is not None and search_string.lower() in node.name.lower():
-                    search_nodes.append(node)
-            self.__update_search_view(search_nodes)
+            search_type = self.__search_dropdown.get()
+            if search_type == "Short Name":
+                search_nodes = []

```

Comment:
maybe, its better if we move the search_nodes = [] outside(before) the if elif loop. We don't need to have duplicate sets inside each of the branches and this set could be added with the value based on the search criteria. And, similarly, self.__update_search_view(search_nodes) can also be called outside with the results obtained from the search criteria.

Reply:
No, I would not say so. It was on purpose to keep separate collections for each if branch (i.e. search type).
(Since if statements don't define a scope in Python. Neither do loops, with statements, try / except, etc. Only modules, functions and classes define scopes.)

Reply:
I changed the code as below and it seems to work. 
```python
search_nodes = []
search_type = self.__search_dropdown.get()
if search_type == "Short Name":
    for node in self.__asr_explorer_id_to_node_dict.values():
        if node.name is not None and search_string.lower() in node.name.lower():
            search_nodes.append(node)

elif search_type == "Autosar Type":
    for node in self.__asr_explorer_id_to_node_dict.values():
        node_type = node.__class__.__name__
        if node_type is not None and search_string.lower() in node_type.lower():
            search_nodes.append(node)

elif search_type == "Regular Expression":
    for node in self.__asr_explorer_id_to_node_dict.values():
        if node.name is not None:
            matched_re = re.search(search_string, node.name)
            if matched_re is not None and matched_re.group()!='':
                search_nodes.append(node)

self.__update_search_view(search_nodes)
```

and it seems to work! Can you check this?

Reply:
Tried it. Made some small tweaks and pushed it to the branch.